### PR TITLE
Deci-ai models onboarding

### PR DESCRIPTION
**Pull request by HrishikeshGeedMS**
**Model assets in preview test1**

DeciLm-7b - https://ml.azure.com/registries/azureml-preview-test1/models/Deci-DeciLM-7B/version/7?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview

DeciLm-7b-instruct - https://ml.azure.com/registries/azureml-preview-test1/models/Deci-DeciLM-7B-instruct/version/7?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview

Decicoder - https://ml.azure.com/registries/azureml-preview-test1/models/Deci-DeciCoder-1b/version/7?tid=72f988bf-86f1-41af-91ab-2d7cd011db47#overview

**Job to save models in sdkstorage**

deci 7b- https://ml.azure.com/experiments/id/9c3f3a3e-7e47-4182-8fbf-e7301c5a55e5/runs/bd39abf4-76db-4f20-90fa-b5dbb01e1351?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/training_rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-as&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

deci 7b instruct-https://ml.azure.com/experiments/id/bf36c0b5-f978-4c4c-9351-b1645fdd06f8/runs/31176906-efb7-4879-b1cf-8ecf225e32bd?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/training_rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-as&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

decicoder-https://ml.azure.com/experiments/id/b9544de1-1445-4e71-a6c8-72895a8f4d4d/runs/14d07da5-996a-488b-be1f-0de3462f4d1c?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/training_rg/providers/Microsoft.MachineLearningServices/workspaces/model-mgmt-as&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Evaluation jobs
Decicoder -    [evaluate-decicoder - Azure AI | Machine Learning Studio](https://ml.azure.com/experiments/id/84efb869-7831-4835-a77a-d6b372d1d4db?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/donotuseordelete-runner-rg/providers/Microsoft.MachineLearningServices/workspaces/runners-workspace-aueast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
   7B -                [evaluate-7b - Azure AI | Machine Learning Studio](https://ml.azure.com/experiments/id/a3415d3d-50c5-4f95-b2ed-6ece1ec8047b?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/donotuseordelete-runner-rg/providers/Microsoft.MachineLearningServices/workspaces/runners-workspace-aueast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)
   7B-Instruct -   [evaluate-7b-instruct - Azure AI | Machine Learning Studio](https://ml.azure.com/experiments/id/7e56b658-4305-486e-9b8b-a26391f46a89?wsid=/subscriptions/ed2cab61-14cc-4fb3-ac23-d72609214cfd/resourceGroups/donotuseordelete-runner-rg/providers/Microsoft.MachineLearningServices/workspaces/runners-workspace-aueast&tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

### CODE REVIEW

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+The Model Card for DeciCoder 1B provides details about a 1 billion parameter decoder-only code completion model developed by Deci. The model was trained on Python, Java, and JavaScript subsets of Starcoder Training Dataset and uses Grouped Query Attention with a context window of 2048 tokens. It was trained using a Fill-in-the-Middle training objective and generated by Deci's proprietary Neural Architecture Search-based technology, AutoNAC. The model is intended for single/multiline code completion from a context window of up to 2048 tokens. The model has limitations as it has undergone training with source code from Python, Java, and JavaScript, and there is no assurance that the resulting code will function as expected. The Model Card provides details on how to use the model, training details, and evaluation results. The model's checkpoints are licensed under the Apache 2.0 license.
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:1]: Lets use HF model card details and not use GPT for summarization

Comment (novaturient95) [reply assets/models/system/Deci-DeciCoder-1b/description.md:1]: refer to: [doc](https://microsoftapc-my.sharepoint.com/:w:/g/personal/ayushmishra_microsoft_com/ETjbEy6poUhBsdysFKduH3IBmrdEe2gP2Yzza3p6COAAhA?e=tegVOv) for more details

File: assets/models/system/Deci-DeciCoder-1b/spec.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: Deci-DeciCoder-1b
+path: ./
+properties:
+  SHA: 59cead352e3b213c9a77097d3d08e89bf467ef7b
+  datasets: bigcode/starcoderdata
+  inference-recommended-sku: Standard_NC12s_v3, Standard_NC24s_v3, Standard_ND40rs_v2, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/spec.yaml:7]: add min-sku for all supported scenarios

File: assets/models/system/Deci-DeciCoder-1b/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+$schema: https://azuremlschemas.azureedge.net/latest/model.schema.json
+name: Deci-DeciCoder-1b
+path: ./
+properties:
+  SHA: 59cead352e3b213c9a77097d3d08e89bf467ef7b
+  datasets: bigcode/starcoderdata
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/spec.yaml:6]: move datasets to tags

File: assets/models/system/Deci-DeciCoder-1b/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  evaluation_compute_allow_list:
+  - Standard_NC12s_v3
+  - Standard_NC24s_v3
+  - Standard_ND40rs_v2
+  - Standard_ND96asr_v4
+  - Standard_ND96amsr_A100_v4
+  SharedComputeCapacityEnabled: ''
+  task: text-generation
+  license: apache-2.0
+  author: deci-ai
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/spec.yaml:29]: Are we going with deci-ai ? Can you confirm with Tina on this?

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+
+## **Inference samples**
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:4]:
```suggestion
# Inference samples
```

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation" target="_blank">text-generation-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation" target="_blank">text-generation-online-endpoint.sh</a>
+Batch |<a href="https://aka.ms/azureml-infer-batch-sdk-text-generation" target="_blank">text-generation-batch-endpoint.ipynb</a>| coming soon
+
+
+## **Model Evaluation**
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:12]:
```suggestion
# Model Evaluation
```

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+Task| Use case| Dataset| Python sample (Notebook)| CLI with YAML
+|--|--|--|--|--|
+Text generation | Text generation | <a href="https://huggingface.co/datasets/cnn_dailymail" target="_blank"> cnn_dailymail </a> | <a href="https://aka.ms/azureml-eval-sdk-text-generation/" target="_blank">evaluate-model-text-generation.ipynb</a> | <a href="https://aka.ms/azureml-eval-cli-text-generation/" target="_blank">evaluate-model-text-generation.yml</a>
+
+
+### Sample inputs and outputs (for real-time inference)
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:19]:
```suggestion
# Sample inputs and outputs
```

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+#### Sample input
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:21]:
```suggestion
### Sample input
```

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+              "top_p": 0.95,
+              "temperature": 0.1,
+              "max_new_tokens": 10,
+              "do_sample": true
+      }
+  }
+}
+```
+
+#### Sample output
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:36]:
```suggestion
### Sample output
```

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+
+### Sample inputs and outputs (for real-time inference)
+
+#### Sample input
+```json
+{
+  "input_data": {
+      "input_string": ["def print_hello_world():"],
+      "parameters":{
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:26]: Please format and add

File: assets/models/system/Deci-DeciCoder-1b/spec.yaml (hunk 4 above)
Comment (HrishikeshGeedMS) [reply assets/models/system/Deci-DeciCoder-1b/spec.yaml:29]: Yes ayush ! Nancy and team has tested on this tag. Manoj was in loop always for this.

File: assets/models/system/Deci-DeciCoder-1b/description.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-online-sdk-text-generation" target="_blank">text-generation-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-online-cli-text-generation" target="_blank">text-generation-online-endpoint.sh</a>
+Batch |<a href="https://aka.ms/azureml-infer-batch-sdk-text-generation" target="_blank">text-generation-batch-endpoint.ipynb</a>| coming soon
+
+
+# Model Evaluation
+
+Task| Use case| Dataset| Python sample (Notebook)| CLI with YAML
+|--|--|--|--|--|
+Text generation | Text generation | <a href="https://huggingface.co/datasets/cnn_dailymail" target="_blank"> cnn_dailymail </a> | <a href="https://aka.ms/azureml-eval-sdk-text-generation/" target="_blank">evaluate-model-text-generation.ipynb</a> | <a href="https://aka.ms/azureml-eval-cli-text-generation/" target="_blank">evaluate-model-text-generation.yml</a>
```
Comment (novaturient95) [inline assets/models/system/Deci-DeciCoder-1b/description.md:16]: update the notebook to point to v2 sample please, You can refer to falcon

**Review by novaturient95 — DISMISSED** [review]

File: assets/models/system/Deci-DeciCoder-1b/description.md (hunk 11 above)
Comment (HrishikeshGeedMS) [reply assets/models/system/Deci-DeciCoder-1b/description.md:8]: Done

**Review by novaturient95 — APPROVED** [review]

**Pull request merged**